src/instance.rs

```diff
@@ -123,7 +124,7 @@ impl<T> Py<T> {
         if ptr.is_null() {
             crate::err::panic_after_error();
         } else {
-            Py::from_owned_ptr(ptr)
```

Comment:
I don't think we have to change this line.
`from_owned_ptr` should behave same as `Nonnull::new~` in release mode.

Reply:
Is this fine now? I'm not sure if you are saying you want `from_owned_ptr` to replace `debug_assert` with a runtime check (where `NonNull::new` checks `ptr.is_null()`).